rs/state_machine_tests/src/tests.rs

```diff
@@ -90,38 +81,18 @@ fn public_derivation_path() {
     let ecdsa_secret_key: PrivateKey =
         PrivateKey::deserialize_sec1(private_key_bytes.as_slice()).unwrap();
 
-    let master_public_key = MasterPublicKey {
-        algorithm_id: AlgorithmId::EcdsaSecp256k1,
-        public_key: ecdsa_secret_key.public_key().serialize_sec1(true),
-    };
-
     let caller = PrincipalId::new_user_test_id(1);
 
-    let extended_derivation_path = ExtendedDerivationPath {
-        caller,
-        derivation_path: vec![],
-    };
-
-    let derivation_path = DerivationPath::new(
-        std::iter::once(caller.as_slice().to_vec())
-            .map(DerivationIndex)
-            .collect::<Vec<_>>(),
-    );
+    let path = DerivationPath::from_canister_id_and_path(caller.as_slice(), &[]);
 
-    let derived_public_key_bytes = derivation_path
-        .public_key_derivation(
-            &ecdsa_secret_key.public_key().serialize_sec1(true),
-            &[0; 32],
-        )
-        .expect("couldn't derive ecdsa public key");
+    let derived_key = ecdsa_secret_key
+        .public_key()
+        .derive_subkey(&path)
+        .0
+        .serialize_sec1(true);
 
     assert_eq!(
-        ic_crypto_utils_canister_threshold_sig::derive_threshold_public_key(
-            &master_public_key,
-            &extended_derivation_path
-        )
-        .expect("failed to derive tecdsa key")
-        .public_key,
-        derived_public_key_bytes.derived_public_key
+        hex::encode(&derived_key),
+        "03fda02786d72d691d807a10a3de60522b664472ec2f06a704cc34ebe2fc26724c"
```

Comment:
How was this hex string generated?

Reply:
I ran the test and printed the value that was created / that the test previously agreed was the "correct" output.

The test as it stood was not as helpful as I think the author intended, because the two crates that were being called are both just wrappers around the same underlying implementation.

Reply:
Makes sense; thank you for the explanation!